### PR TITLE
feat KHP3-7123: Aligned Mortuary Workflow and added Autopsy Report Form

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
The mortuary workflow has been streamlined to ensure proper alignment from the initial "Awaiting Queue" stage to "Admitted" and finally to "Discharged." Additionally, an Autopsy Report Form has been integrated into the workflow to support detailed reporting for deceased patients. I believe the mortuary workflow is complete

## Screenshots
morgue-updated

*None.*

## Related Issue
see: https://thepalladiumgroup.atlassian.net/browse/KHP3-7123
*None.*

## Other

*None.*

### CODE REVIEW

**Comment by makombe** [comment]
@its-kios09 Thanks for the good work. I think we need to be consistent with UIs. What if you just show the counts under the respective tabs something similar to the way we have done for lab,procedures etc. rather than having the counts on the header?

**Comment by its-kios09** [comment]
@makombe, Thanks a lot for the feedback. Let me update the PR

**Comment by ojwanganto** [comment]
@makombe i see what you meant. Unlike the ward app, we have tabs corresponding to the metrics. I support the proposal to make this consistent with other implementations.

**Comment by its-kios09** [comment]
@makombe @ojwanganto I have reverted to  we have tabs corresponding to the metrics, Thanks for the feedback

https://github.com/user-attachments/assets/d49e2853-31ec-4c18-8864-70e093805f78

File: packages/esm-morgue-app/src/tables/admitted-queue.component.tsx [hunk 1]
```diff
@@ -32,6 +32,15 @@ export const AdmittedQueue: React.FC = () => {
     return <ErrorState error={error} headerTitle={t('allocation', 'Allocation')} />;
   }
 
+  if (admittedPatients?.length === 0) {
+    return (
+      <EmptyDeceasedSearch
+        title={t('allocations', 'Allocation')}
+        subTitle={t('noAdmittedPersons', 'There are no admitted bodies on the any compartments')}
```
Comment (ojwanganto) [inline packages/esm-morgue-app/src/tables/admitted-queue.component.tsx:39]: Is the working here correct? Can we omit the work 'compartment' and for for 'There are no admitted bodies'?
Also, I don't think EmptyDeceasedSearch is a good name for the component. I guess it should be something like EmptyMorgueAdmission or something related. Am just thinking :)

File: packages/esm-morgue-app/src/tables/waiting-queue.component.tsx [hunk 2]
```diff
@@ -46,11 +52,11 @@ export const WaitingQueue: React.FC = () => {
 
   const rows = awaitingPatients.map((patient) => ({
     id: patient.uuid,
-    name: toUpperCase(patient.person.display),
+    name: patient.person.display?.toUpperCase(),
     gender: patient.person.gender,
     age: patient?.person?.age,
     identifier: patient?.identifiers[0]?.identifier,
-    deathDate: formatDate(new Date(patient.person.deathDate)),
+    deathDate: new Date(patient.person.deathDate).toLocaleString(),
```
Comment (ojwanganto) [inline packages/esm-morgue-app/src/tables/waiting-queue.component.tsx:59]: Let's check that the death date is not null.

**Comment by its-kios09** [comment]
@ojwanganto Thanks for the left suggestion. Let me work on them

**Comment by ojwanganto** [comment]
Also, we should paginate the requests so that the tabs don't take long to load. We can load just a small set and load the rest on demand. The search should also initiate a call to the backend if an item is not found in the small set of results available on the frontend

File: packages/esm-morgue-app/src/tables/admitted-queue.component.tsx (hunk 1 above)
Comment (its-kios09) [reply packages/esm-morgue-app/src/tables/admitted-queue.component.tsx]: @ojwanganto, you made a good point about improving semantics, I have changed the naming and phrasing as per your suggestion

File: packages/esm-morgue-app/src/tables/waiting-queue.component.tsx (hunk 2 above)
Comment (its-kios09) [reply packages/esm-morgue-app/src/tables/waiting-queue.component.tsx]: I have resolved

**Comment by its-kios09** [comment]
> Also, we should paginate the requests so that the tabs don't take long to load. We can load just a small set and load the rest on demand. The search should also initiate a call to the backend if an item is not found in the small set of results available on the frontend

Absolutely, @ojwanganto, Thanks that's a great suggestion, Paginating the requests will significantly improve the performance and user experience. Let me work on it

**Comment by its-kios09** [comment]
@ojwanganto I have successfully implemented server-side pagination. The only remaining task is to wait for @PatrickWaweru to update the endpoint to support server-side search. Currently, Waweru's implementation searches for patient's property dead = true.

**Review by makombe — DISMISSED** [review]
@donaldkibet Please add your input here. Thanks

**Comment by its-kios09** [comment]
@donaldkibet kindly I was requesting your review

**Review by makombe — APPROVED** [review]
@its-kios09 let merge this any other issue you can address in a follow up PR. Thanks

**Pull request merged**